### PR TITLE
(fix): Fixed Pagination Logic in Medical History Panel of Morgue Module

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Screenshots
encounter

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-morgue-app/src/panels/medical-history.component.tsx [hunk 1]
```diff
@@ -81,15 +80,19 @@ const MedicalHistoryView: React.FC = () => {
...
-  const rows = filteredEncounters?.map((encounter, idx) => {
-    return {
-      id: `encounter-${idx}`,
-      encounterDatetime: formatDate(new Date(encounter.encounterDatetime)),
-      visitType: encounter.visitType,
-      encounterType: encounter.encounterType,
-      formName: encounter.form,
-    };
-  });
+  const paginatedEncounterItems = useMemo(() => {
```
Comment (donaldkibet) [inline packages/esm-morgue-app/src/panels/medical-history.component.tsx:83]: would recommend use of server side pagination `useServerPagination` https://github.com/openmrs/openmrs-esm-core/blob/5a6bbe2d83ae675e624a1e9bf332dcda0394cb86/packages/framework/esm-react-utils/src/useOpenmrsPagination.ts#L107

File: packages/esm-morgue-app/src/panels/medical-history.component.tsx [hunk 2]
```diff
@@ -186,26 +189,20 @@ const MedicalHistoryView: React.FC = () => {
...
-              }
-            }}
-          />
-        )}
+        <Pagination
+          forwardText="Next page"
+          backwardText="Previous page"
+          page={currentPage}
+          pageSize={pageSize}
+          pageSizes={[5, 10, 20, 50, 100]}
```
Comment (donaldkibet) [inline packages/esm-morgue-app/src/panels/medical-history.component.tsx:197]: This should not be hard coded

**Review by donaldkibet — CHANGES_REQUESTED** [review]

**Comment by its-kios09** [comment]
@donaldkibet thanks for suggestions

**Comment by donaldkibet** [comment]
cc @its-kios09 seems we have some merge conflicts

File: packages/esm-morgue-app/src/panels/medical-history.component.tsx (hunk 1 above)
Comment (its-kios09) [reply packages/esm-morgue-app/src/panels/medical-history.component.tsx]: Sorted

**Comment by its-kios09** [comment]
@donaldkibet i have sorted the pagination let work on the conflicts

**Comment by donaldkibet** [comment]
Seems we still have conflicts 
Screenshot 2024-10-16 at 10 27 51

**Comment by its-kios09** [comment]
@donaldkibet I have sorted the conflicts

**Comment by its-kios09** [comment]
@donaldkibet Thanks

File: packages/esm-morgue-app/src/card/compartment.scss [hunk 3]
```diff
@@ -1,7 +1,7 @@
 @use '@carbon/layout';
 @use '@carbon/type';
 @use '@carbon/colors';
-@use '~@openmrs/esm-styleguide/src/vars' as *;
+// @use '~@openmrs/esm-styleguide/src/vars' as *;
```
Comment (donaldkibet) [inline packages/esm-morgue-app/src/card/compartment.scss:4]: We should remove unused imports

File: packages/esm-morgue-app/src/details/deceased-details.scss [hunk 4]
```diff
@@ -1,7 +1,7 @@
 @use '@carbon/layout';
 @use '@carbon/type';
 @use '@carbon/colors';
-@import '~@openmrs/esm-styleguide/src/vars';
+// @import '~@openmrs/esm-styleguide/src/vars';
```
Comment (donaldkibet) [inline packages/esm-morgue-app/src/details/deceased-details.scss:4]: Same here

**Review by donaldkibet — CHANGES_REQUESTED** [review]

**Review by donaldkibet — APPROVED** [review]

**Pull request merged**